components/lcd_touch/esp_lcd_touch_gt1151/esp_lcd_touch_gt1151.c

```diff
@@ -0,0 +1,220 @@
+    ESP_GOTO_ON_FALSE(gt1151, ESP_ERR_NO_MEM, err, TAG, "Touch handle malloc failed");
+
+    /* Communication interface */
+    gt1151->io = io;
+    /* Only supported callbacks are set */
+    gt1151->read_data = read_data;
+    gt1151->get_xy = get_xy;
+    gt1151->del = del;
+    /* Mutex */
+    gt1151->data.lock.owner = portMUX_FREE_VAL;
+    /* Save config */
+    memcpy(&gt1151->config, config, sizeof(esp_lcd_touch_config_t));
+
+    /* Prepare pin for touch interrupt */
+    if (gt1151->config.int_gpio_num != GPIO_NUM_NC) {
+        const gpio_config_t int_gpio_config = {
+            .mode = GPIO_MODE_INPUT,
+            .pin_bit_mask = BIT64(gt1151->config.int_gpio_num)
+        };
+        ESP_GOTO_ON_ERROR(gpio_config(&int_gpio_config), err, TAG, "GPIO intr config failed");
+    }
+    /* Prepare pin for touch controller reset */
+    if (gt1151->config.rst_gpio_num != GPIO_NUM_NC) {
+        const gpio_config_t rst_gpio_config = {
+            .mode = GPIO_MODE_OUTPUT,
+            .pin_bit_mask = BIT64(gt1151->config.rst_gpio_num)
+        };
+        ESP_GOTO_ON_ERROR(gpio_config(&rst_gpio_config), err, TAG, "GPIO reset config failed");
+    }
+    /* Reset controller */
+    ESP_GOTO_ON_ERROR(reset(gt1151), err, TAG, "Reset failed");
+    /* Read product id */
+    read_product_id(gt1151);
+    *tp = gt1151;
+
+    return ESP_OK;
+err:
+    if (gt1151) {
+        del(gt1151);
+    }
+    ESP_LOGE(TAG, "Initialization failed!");
+    return ret;
+}
+
+static esp_err_t read_data(esp_lcd_touch_handle_t tp)
+{
+    uint8_t buf[DATA_BUFF_LEN(MAX_TOUCH_NUM)];
+    ESP_RETURN_ON_ERROR(i2c_read_bytes(tp, READ_XY_REG, buf, 1), TAG, "I2C read failed!");
+
+    /* Any touch data? */
+    if ((buf[0] & 0x80) == 0x00) {
+        i2c_write_byte(tp, READ_XY_REG, 0);
+    } else {
+        uint8_t touch_cnt = 0;
+        /* Get count of touched points */
+        touch_cnt = buf[0] & 0x0f;
+        if (touch_cnt > MAX_TOUCH_NUM || touch_cnt == 0) {
+            i2c_write_byte(tp, READ_XY_REG, 0);
+            return ESP_OK;
+        }
+        /* Read all points */
+        ESP_RETURN_ON_ERROR(i2c_read_bytes(tp, READ_XY_REG + 1, buf + 1, DATA_BUFF_LEN(touch_cnt) - 1), TAG, "I2C read failed");
+        /* Clear all */
+        i2c_write_byte(tp, READ_XY_REG, 0);
+
+        /* Caculate checksum */
+        uint8_t checksum = 0;
+        for (int i = 0; i < DATA_BUFF_LEN(touch_cnt); i++) {
+            checksum += buf[i];
+        }
+        ESP_RETURN_ON_FALSE(!checksum, ESP_ERR_INVALID_CRC, TAG, "Checksum error");
+
+        taskENTER_CRITICAL(&tp->data.lock);
+        /* Expect Number of touched points */
+        touch_cnt = (touch_cnt > CONFIG_ESP_LCD_TOUCH_MAX_POINTS ? CONFIG_ESP_LCD_TOUCH_MAX_POINTS : touch_cnt);
+        tp->data.points = touch_cnt;
+
+        /* Fill all coordinates */
+        for (int i = 0; i < touch_cnt; i++) {
+            tp->data.coords[i].x = ((uint16_t)buf[(i * 8) + 3] << 8) + buf[(i * 8) + 2];
```

Comment:
What about use packed structure for get data. For example like here https://github.com/espressif/esp-bsp/blob/master/components/lcd_touch/esp_lcd_touch_tt21100/esp_lcd_touch_tt21100.c line 114.

Reply:
Of course, please check the new [commit](https://github.com/espressif/esp-bsp/pull/75/commits/eab1a018c105292926eef02142db34ab2e4a2bfc).